packages/taquito-michelson-encoder/src/schema/storage.ts

```diff
@@ -165,6 +166,9 @@ export class Schema {
     if (this.root instanceof TicketDeprecatedToken && val.ticketer && val.value && val.amount) {
       return true;
     }
+    if (this.root instanceof MapToken && this.root.ValueSchema instanceof BigMapToken) {
```

Comment:
I understand the MapToken check, but why do we need to also check `this.root.ValueSchema`?

Reply:
so far map typecheck as been fine its just when checking the type of a BigMapToken

Reply:
nested inside